### PR TITLE
ratchet: init at 0.7.0

### PR DESCRIPTION
**Pull request by ryanccn**
## Description of changes

https://github.com/sethvargo/ratchet

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ryanccn** [comment]
Result of `nixpkgs-review pr 282464` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ratchet</li>
  </ul>
</details>

**Comment by getchoo** [comment]
Result of `nixpkgs-review pr 282464` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ratchet</li>
  </ul>
</details>

checkPhase fails on aarch64-linux due to one of the tests attempting to make a connection to github
```
Running phase: checkPhase
ok      github.com/sethvargo/ratchet/command    0.043s
ok      github.com/sethvargo/ratchet/internal/atomic    0.033s
ok      github.com/sethvargo/ratchet/parser     0.058s
--- FAIL: TestResolve (0.00s)
    --- FAIL: TestResolve/path (0.02s)
        actions_test.go:45: failed to get commit sha: Get "https://api.github.com/repos/github/codeql-action/commits/v1": dial tcp: lookup api.github.com on [::1]:53: read udp [::1]:53089>
    --- FAIL: TestResolve/default (0.02s)
        actions_test.go:45: failed to get commit sha: Get "https://api.github.com/repos/actions/checkout/commits/v3": dial tcp: lookup api.github.com on [::1]:53: read udp [::1]:53089->[:>
FAIL
FAIL    github.com/sethvargo/ratchet/resolver   0.069s
FAIL
```

File: pkgs/by-name/ra/ratchet/package.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "-X=github.com/sethvargo/ratchet/internal/version.version=${version}"
+    "-X=github.com/sethvargo/ratchet/internal/version.commit=${src.rev}"
+  ];
+
+  # skip tests that require network access
+  checkFlags =
+    let
+      skippedTests = [ "TestResolve" ];
+    in
+    [ "-skip=^${builtins.concatStringsSep "$|^" skippedTests}$" ];
```
Comment (getchoo) [inline pkgs/by-name/ra/ratchet/package.nix:30]: i don't think we need to be this careful with the regex, plus using two separate strings makes things a bit neater. the comment about network access should also only apply to the test it's referring to, not any test that may be skipped

```suggestion
  checkFlags =
    let
      skippedTests = [
        "TestResolve" # requires network access
      ];
    in
    [ "-skip" (lib.concatStringsSep "|" skippedTests) ];
```

**Review by getchoo — CHANGES_REQUESTED** [review]

**Comment by getchoo** [comment]
Result of `nixpkgs-review pr 282464` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ratchet</li>
  </ul>
</details>

**Review by getchoo — APPROVED** [review]

File: pkgs/by-name/ra/ratchet/package.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        "TestLatestVersion" # requires network access
+      ];
+    in
+    [ "-skip" (lib.concatStringsSep "|" skippedTests) ];
+
+  meta = with lib; {
+    description = "A tool for securing CI/CD workflows with version pinning";
+    homepage = "https://github.com/sethvargo/ratchet";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ryanccn ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ra/ratchet/package.nix:38]:
```suggestion
    maintainers = with maintainers; [ ryanccn ];
    mainProgram = "ratchet";
```

**Comment by ryanccn** [comment]
@cameronraysmith I'm happy to go with your PR moving forward! Do add me as a maintainer there, thanks.

**Comment by ryanccn** [comment]
Superseded by #311799.

**Pull request closed without merging**